src/main/java/spoon/reflect/code/CtTypeAccess.java

```diff
@@ -16,13 +16,47 @@
  */
 package spoon.reflect.code;
 
+import spoon.reflect.reference.CtTypeReference;
+
 /**
  * This code element defines an access to a type.
  *
  * In Java, it is generally of the form: <code>Type</code>.
  *
+ * For example, you can have:
+ *
+ * <pre>
+ *     Type.field
+ *     Type.method()
+ *     Type::method
+ *     t instanceof Type
+ *     Type.class
+ * </pre>
+ *
  * @param <T>
  * 		type in the expression.
  */
 public interface CtTypeAccess<T> extends CtExpression<T> {
+	/**
+	 * Returns type represented and contained in the type access.
+	 *
+	 * @return CtTypeReference.
+	 */
+	CtTypeReference<T> getAccessedType();
```

Comment:
T represents the type of the element (Void), not the accessed type.

--Martin


On 01/05/2016 05:41 PM, Gérard Paligot wrote:
>
> In src/main/java/spoon/reflect/code/CtTypeAccess.java
> <https://github.com/INRIA/spoon/pull/470#discussion_r48865190>:
>
> >   * @param <T>
> >   * 		type in the expression.
> >   */
> >  public interface CtTypeAccess<T> extends CtExpression<T> {
> > +	/**
> > +	 * Returns type represented and contained in the type access.
> > +	 *
> > +	 * @return CtTypeReference.
> > +	 */
> > +	CtTypeReference<T> getAccessedType();
>
> Why?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/INRIA/spoon/pull/470/files#r48865190>.
>